File: frontend/src/api/exchangeRates.ts
import { apiGet } from "./client";
import type { ApiResponse } from "../types/api";

export interface ExchangeRate {
  rate_date: string;
  from_currency: string;
  to_currency: string;
  rate: number;
  source: string;
}

export function fetchCurrentRate(): Promise<ApiResponse<ExchangeRate | null>> {
  return apiGet("/api/v1/exchange-rates/current");
}

File: src/api/schemas/exchange_rates.py
"""Pydantic schemas for exchange rate endpoints."""

from __future__ import annotations

from datetime import date, datetime
from decimal import Decimal

from pydantic import BaseModel


class ExchangeRateSchema(BaseModel):
    rate_date: date
    from_currency: str = "USD"
    to_currency: str = "BRL"
    rate: Decimal
    source: str = "bcb_ptax"
    created_at: datetime | None = None


class ExchangeRateHistorySchema(BaseModel):
    rates: list[ExchangeRateSchema]
    count: int
